### ISSUE TITLE
Kubernetes cannot recover after losing the connection

### ISSUE THREAD

**Issue by axel7083** [containers/podman-desktop#9156]
### Bug description

While working on https://github.com/containers/podman-desktop/issues/8341 I noticed a problem when podman desktop lose the connection it is **never able** to recover (on branch main 01/10/2024).

https://github.com/user-attachments/assets/577789f3-296e-4031-a3a1-2f63c15902f3

### Operating system

windows 11

### Installation Method

Other

### Version

next (development version)

### Steps to reproduce

1. create kind cluster
2. assert node visible in kubernetes page
3. stop kind container
4. assert connection lost
5. start kind container 
6. assert nothing is hapening

The only way to reconnect is to close podman desktop and restart it.

### Relevant log output

```shell
17:36:19 [main] Trying to watch pods on the kubernetes context named "kind-kind-cluster" but got a connection refused, retrying the connection in 18s. Error: Forbidden)

17:37:22 [main] Kube deployment watch ended AbortError: The user aborted a request.
17:37:28 [main] Trying to watch routes on the kubernetes context named "kind-kind-cluster" but got a connection refused, retrying the connection in 84s. Error: HTTP-Code: 404
Message: Unknown API Status Code!
Body: "404 page not found\n"
Headers: {"audit-id":"51677f0f-4cab-48ab-9825-6ee0a846f25c","cache-control":"no-cache, private","connection":"close","content-length":"19","content-type":"text/plain; charset=utf-8","date":"Tue, 01 Oct 2024 15:37:28 GMT","x-content-type-options":"nosniff","x-kubernetes-pf-flowschema-uid":"1cc99633-0cfc-4d6b-8aa5-142664664a08","x-kubernetes-pf-prioritylevel-uid":"996d21c0-4bbf-42df-b078-33bcf6bfab1c"})

17:38:52 [main] Trying to watch routes on the kubernetes context named "kind-kind-cluster" but got a connection refused, retrying the connection in 84s. Error: HTTP-Code: 404
Message: Unknown API Status Code!
Body: "404 page not found\n"
Headers: {"audit-id":"bd2d8ccd-132a-47b5-a261-09be85016af3","cache-control":"no-cache, private","connection":"close","content-length":"19","content-type":"text/plain; charset=utf-8","date":"Tue, 01 Oct 2024 15:38:52 GMT","x-content-type-options":"nosniff","x-kubernetes-pf-flowschema-uid":"1cc99633-0cfc-4d6b-8aa5-142664664a08","x-kubernetes-pf-prioritylevel-uid":"996d21c0-4bbf-42df-b078-33bcf6bfab1c"})
```

### Additional context

I am able a few seconds after restarting the kind container to access `https://127.0.0.1:60143/version`.

**Comment by axel7083** [comment]
cc @feloy

**Comment by jeffmaury** [comment]
The message is confusing here: we are trying ro retrieve OpenShift routes on a Kind cluster thus the 404

**Comment by feloy** [comment]
Are you restarting Podman Desktop when the cluster is stopped (between 4 and 5)?

In this specific case (when the current cluster is not reachable at startup), it seems that it is not implemented (the "secondary" informers - the informers for other resources than Pods and Deployments - and never started)

I can reproduce from time to time the problem without restarting Podman Desktop, but I need to investigate more to see if the secondary informers are stopped (and not restarted) after some unreachable time.

**Comment by axel7083** [comment]
> Are you restarting Podman Desktop when the cluster is stopped (between 4 and 5)?

No, as visible in the video I did not restarted podman desktop. I stop the kind container from the cli, wait until the connection is lost, then start the container.

**Comment by feloy** [comment]
Opened an issue https://github.com/kubernetes-client/javascript/issues/1933 which seems to be the cause

**Comment by odockal** [comment]
Issue has status Done but it is not closed, was it resolved?
@feloy @axel7083 can you folks close/retest, please?

**Comment by feloy** [comment]
Fixed by #9409
